pythonic-query.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+import time
+import subprocess
+import json
+import requests
+
+studyid_counter = 0
+formatted_datetime = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+
+def clean_text(text):
+    if isinstance(text, str):
+        text = text.replace(" :", "").replace(": ", "").replace(":", "")  # remove misplaced colons
+        text = text.replace('"', "'")  # replace double quotes with single quotes
+        text = text.lstrip(", ")  # remove leading commas and spaces
+        if text.startswith(">"):  # ensure '>' values are properly quoted
+            text = f'"{text}"'
+    return text
+
+
+# remove duplicate keys in properties
+def remove_duplicates(properties):
+    unique_properties = {}
+    for prop_name, prop_values in properties.items():
+        unique_properties[prop_name] = list(set(prop_values))
+    return unique_properties
+
+
+# fetch the list of experiment accessions dynamically from GXA API
+def get_experiment_ids():
+    bash_command = 'curl -sS https://www.ebi.ac.uk/gxa/json/experiments | jq -r ".experiments[].experimentAccession" | sort -u'
+    experiment_accessions = subprocess.check_output(bash_command, shell=True, text=True)
+    study_ids = experiment_accessions.strip().split("\n")
+
+    return study_ids
+
+
+def fetch_and_parse_data(study_id):
+    study_url = f"https://www.ebi.ac.uk/gxa/json/experiments/{study_id}"
+
+    max_retries = 5
+    retry_delay = 5
```

Comment:
It might be worth moving these two to the top as sometimes this needs changing.. so easy to spot